### PR TITLE
add beta annotations to apps

### PR DESCRIPTION
**Pull request by deads2k**
This adds the lifecycle annotation tags matching https://github.com/kubernetes/enhancements/blob/master/keps/sig-architecture/1635-prevent-permabeta/README.md 

The clock is set for removal of beta levels in 1.22.

/kind cleanup
/priority important-soon
@kubernetes/sig-apps-misc @kubernetes/sig-api-machinery-misc 

```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/api/apps/v1beta1/types.go [hunk 1]
```diff
@@ -56,6 +56,10 @@ type ScaleStatus struct {
 }
 
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:prerelease-lifecycle-gen:introduced=1.6
+// +k8s:prerelease-lifecycle-gen:deprecated=1.8
+// +k8s:prerelease-lifecycle-gen:removed=1.22
```
Comment (liggitt) [inline staging/src/k8s.io/api/apps/v1beta1/types.go:61]: all apps/v1beta1 and apps/v1beta2 types are unserved as of 1.18 https://github.com/kubernetes/kubernetes/pull/85903

Comment (deads2k) [reply staging/src/k8s.io/api/apps/v1beta1/types.go:61]: > all apps/v1beta1 and apps/v1beta2 types are unserved as of 1.18 #85903

Look at me, being too generous

File: staging/src/k8s.io/api/apps/v1beta1/types.go [hunk 2]
```diff
@@ -355,6 +371,10 @@ type DeploymentSpec struct {
 }
 
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:prerelease-lifecycle-gen:introduced=1.6
+// +k8s:prerelease-lifecycle-gen:deprecated=1.8
+// +k8s:prerelease-lifecycle-gen:removed=1.18
+// +k8s:prerelease-lifecycle-gen:replacement=apps,v1,DeploymentRollback
```
Comment (liggitt) [inline staging/src/k8s.io/api/apps/v1beta1/types.go:377]: no replacement for this API, support was removed in later versions

**Comment by liggitt** [comment]
/lgtm
/approve
/retest

one nit https://github.com/kubernetes/kubernetes/pull/90457/files#r416015663, follow up to drop that is fine

**Pull request merged**